File: week6/ex2_taeho.js
function solution(order) {
    
    const reverseOrder = [...order].reverse();
    const stack = [];
    let count = 0;

    for (let i = 1; i <= order.length; i++) {
        stack.push(i);
        while (stack.length && stack.at(-1) === reverseOrder.at(-1)) {
            count += 1;
            stack.pop();
            reverseOrder.pop();
        }
    }
    return count;
}

File: week9/ex2_taeho.py
from collections import Counter

def solution(k, tangerine):
    
    counter = Counter(tangerine)
    counts = [count for count in Counter(tangerine).values()]
    counts.sort(reverse=True)
    
    i = 0
    while(k > 0 and i <len(counts)):
        k = k - counts[i]
        i += 1
    
    
    return i